aleph/views/entities_api.py

```diff
@@ -576,3 +577,83 @@ def expand(entity_id):
             "results": results,
         }
     )
+
+
+@blueprint.route("/api/2/entities/<entity_id>/entitysets", methods=["GET"])
+def entity_entitysets(entity_id):
+    """Returns a list of entitysets which the entity has references in
+    ---
+    get:
+      summary: Shows EntitySets which reference the given entity
+      description: >-
+        Search for all entitysets which reference the given entity. The entity
+        sets can be filtered based on it's collection id, label, type or the
+        judgement of the entity within the set.
+      parameters:
+      - in: path
+        name: entity_id
+        required: true
+        schema:
+          type: string
+      - in: query
+        name: 'filter:type'
+        description: Restrict to a EntitySets of a particular type
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:label'
+        description: Restrict to a EntitySets with a particular label
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:judgement'
+        description: Restrict to a specific profile judgement
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:collection_id'
+        description: Restrict to entity sets within particular collections
+        schema:
+          items:
+            type: string
+          type: array
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/EntitySet'
+          description: OK
+      tags:
+      - Entity
+      - Profiles
+    """
+    entity = get_index_entity(entity_id, request.authz.READ)
+
+    parser = QueryParser(request.args, request.authz)
+    collection_ids = [
+        cid
+        for cid in parser.filters.get("collection_id", [])
+        if request.authz.can(cid, request.authz.READ)
+    ]
```

Comment:
If I read this correctly, then having no `filter:collection_id` filters will return an empty array, and in `.by_entity_id` that will disable the authorisation filter.

Reply:
Correct, but then `EntitySetItemsQuery` will filter out based on the `collection_id`. However, at that point the data is already fetched, so I can also add a:

```
if not collection_ids:
    collection_ids = request.authz.collections(request.authz.READ)
```

Reply:
Actually, hold up. I'm confusing this part of code with what I was looking at for #1429. `EntitySetItemsQuery` has no bearing on this.

Reply:
Yeah I think that would be cool. I don't think `EntitySetItemsQuery` is involved?

Reply:
For #1429 I was making sure to myself that `GET /entitysets/XXX/entities` wouldn't leak which lead my down a rabbithole.